src/data.c

```diff
@@ -1060,6 +1070,9 @@ static void poll_ctrl(void) {
         }
 
         destroy_pool(cmd->pool);
+        session.curr_cmd = curr_cmd;
+        session.curr_cmd_id = curr_cmd_id;
+        session.curr_cmd_rec = curr_cmd_rec;
```

Comment:
Here we restore the pointers for the previously running data transfer command, and effectively clear out the now-stale pointers to the interrupting command.